ext/openssl/openssl.c

```diff
@@ -703,7 +706,13 @@ static void add_assoc_name_entry(zval * val, char * key, X509_NAME * name, int s
 				add_assoc_stringl(&subitem, sname, (char *)to_add, to_add_len);
 			}
 		}
+
+		if (needs_free) {
+			/* ASN1_STRING_to_UTF8(3): The buffer out should be freed using free(3) */
+			free(to_add);
```

Comment:
This should be actually OPENSSL_free as ASN1_STRING_to_UTF8 uses ASN1_mbstring_copy which uses OPENSSL_malloc . This is also documented in

https://www.openssl.org/docs/man1.0.2/crypto/ASN1_STRING_to_UTF8.html

It's not an issue now when we use default allocator but might be problematic if we switch allocator for OpenSSL (e.g. emalloc, efree...). It might need a bit more thinking though but this should be fixed anyway as it's also cleaner from the code PoV...

@lt or @krakjoe  If you are not busy please change it. Otherwise I will do that later once I get some time...